File: src/main/java/no/nav/foreldrepenger/stønadskonto/regelmodell/StønadskontoResultat.java
package no.nav.foreldrepenger.stønadskonto.regelmodell;

import java.util.Collections;
import java.util.EnumMap;
import java.util.Map;
import java.util.Objects;

public class StønadskontoResultat {

    private final Map<StønadskontoKontotype, Integer> stønadskontoer = new EnumMap<>(StønadskontoKontotype.class);

    public StønadskontoResultat(Map<StønadskontoKontotype, Integer> stønadskontoer) {
        Objects.requireNonNull(stønadskontoer);
        this.stønadskontoer.putAll(stønadskontoer);
    }

    public Map<StønadskontoKontotype, Integer> getStønadskontoer() {
        return Collections.unmodifiableMap(stønadskontoer);
    }

}

File: stonadskonto/src/StønadskontoResultat.ts
import { StønadskontoKontotype } from './StønadskontoKontotype';

export class StønadskontoResultat {
  private stønadskontoer: Map<StønadskontoKontotype, number>;

  constructor(stønadskontoer: Map<StønadskontoKontotype, number>) {
    if (!stønadskontoer) {
      throw new Error('stønadskontoer cannot be null');
    }
    this.stønadskontoer = new Map(stønadskontoer);
  }

  getStønadskontoer(): Map<StønadskontoKontotype, number> {
    return new Map(this.stønadskontoer);
  }
}

